include/onnxruntime/core/graph/indexed_sub_graph.h

```diff
@@ -54,19 +54,28 @@
 
   /** Set the meta definition needed to represent this subgraph as a FunctionProto
   It's needed IF AND ONLY IF there are multiple indexes contained in #nodes. */
-  void SetMetaDef(std::unique_ptr<MetaDef>&& meta_def_) {
-    meta_def = std::move(meta_def_);
+  void SetMetaDef(std::unique_ptr<MetaDef>&& meta_def) {
+    meta_def_ = std::move(meta_def);
   }
 
-  /** Gets the meta definition needed to represent this subgraph as a FunctionProto. 
+  /** Gets the meta definition needed to represent this subgraph as a FunctionProto.
   @returns MetaDef instance if it has been set. nullptr if not. */
   const MetaDef* GetMetaDef() const {
-    return meta_def.get();
+    return meta_def_.get();
   }
 
+  // use an existing schema instead of generating one when fusing nodes using the MetaDef.
+  // MetaDef.domain + MetaDef.name => the domain.op_type that a schema must exist for with a valid since_version.
+  void SetUseExistingSchema(bool value) {
+    use_existing_schema_ = value;
+  }
+
+  bool UseExistingSchema() const { return use_existing_schema_; }
+
  private:
   // subgraph meta definition.
-  std::unique_ptr<MetaDef> meta_def;
+  std::unique_ptr<MetaDef> meta_def_;
```

Comment:
if meta_def_ and use_existing_schema_ both have getters and setters, do they need to be private?

Reply:
The bool for using an existing schema could be public. I'll change that.

Could go either way for the MetaDef. As we own the memory having slightly more control over the instance may be desired (e.g. ensure it isn't freed unexpectedly) but not sure it is important here. SetMetaDef is baked into the provider interface though so for now I will leave that as is.